src/Symfony/Component/String/AbstractUnicodeString.php

```diff
@@ -444,6 +444,17 @@ public function width(bool $ignoreAnsiDecoration = true): int
         return $width;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function reverse(): parent
+    {
+        $str = clone $this;
+        $str->string = implode('', array_reverse($str->chunk(1)));
```

Comment:
I think it would make more sense to always respect grapheme clusters, thus use `preg_split(`/\X/u`)` in this method

Reply:
That makes sense indeed 👍